add loader package

HaseemUlHaq opened the pull request:
PR refers to #28 
![Capture](https://user-images.githubusercontent.com/68631902/99563139-c8a0ce80-29ea-11eb-9935-14abedaf8d1d.JPG)

moshthepitt reviewed (changes requested):
Hmmm.  Why does this PR have changes that seem related e.g. changes to the store, ducks, etc.  Can we refactor this PR to only include changes that relate to the Loader component?

HaseemUlHaq commented:
> Hmmm. Why does this PR have changes that seem related e.g. changes to the store, ducks, etc. Can we refactor this PR to only include changes that relate to the Loader component?

@moshthepitt 
Reason for this was that the base branch was `user-management-modularization` but because the branch was merged and deleted the base auto changed to `master`. the loader-reafctor was in old state.
Fixed now.

kelvin-muchiri commented on packages/loaders/package.json:
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@onaio/loaders",
```
I think the name here should be **@opensrp/loaders**?

moshthepitt reviewed (changes requested):
@HaseemUlHaq @kelvin-muchiri we already have a loaders package [here](https://github.com/onaio/js-tools/pull/167).  I do not want us to have two such packages.

kelvin-muchiri commented on client/src/components/page/Loading/tests/__snapshots__/index.test.tsx.snap:
```diff
@@ -1,101 +1,119 @@
+</Loader>
 `;
 
 exports[`components/Loading renders the Loading component when props are given 1`] = `
```
With regard to https://kentcdodds.com/blog/effective-snapshot-testing we can get rid of this snapshot since in this case we should only be interested in testing the props passed to `Ripple` which has been done below, the assumption being that the component `Ripple` should have been tested by the author and we should not worry about its implementation

kelvin-muchiri commented:
> @HaseemUlHaq @kelvin-muchiri we already have a loaders package here. I do not want us to have two such packages.

@moshthepitt I now recall this issue. The changes were to be done [here](https://github.com/onaio/js-tools/tree/master/packages/Loaders) and this repo updated to use the new released version, correct?

moshthepitt commented:
> I now recall this issue. The changes were to be done here and this repo updated to use the new released version, correct?

Yes :+1:

kelvin-muchiri commented:
@HaseemUlHaq We should remove the  **loaders** package from this repo and use the one from **@onaio/loaders** The change @KipSigei suggested was to be done at [JS Tools loaders](https://github.com/onaio/js-tools/tree/master/packages/Loaders.). Clone the repo then once your PR is merged, a release will be made and then you can update the version [here](https://github.com/OpenSRP/web/blob/master/client/package.json#L15)

HaseemUlHaq commented:
Coverage will be increased once #90 will be done,

kelvin-muchiri commented:
@HaseemUlHaq Resolve the conflicts. Since **dist**  dir are now ignored, you may need to remove the dist folder from cache by running `git rm -r --cached packages/keycloak-user-management/dist`

kelvin-muchiri commented on client/src/components/Logout/tests/__snapshots__/index.test.tsx.snap:
```diff
@@ -2,51 +2,60 @@
 
 exports[`components/Logout renders logout component 1`] = `
```
We do not required this snapshot. We only need to check if `Ripple` is present.

kelvin-muchiri commented:
@HaseemUlHaq Coming to think of it, I thought the `Ripple` from `@onaio/loaders` was not configurable. Since we would like to be consistent with those configurations throughout the project , we will have to move this into a package

kelvin-muchiri commented on client/src/components/page/Loading/index.tsx:
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
```
Do we need this component, `client/src/components/page/Loading/index.tsx` ? Since we are now using `@opensrp/loader` we should remove this

haseemVD replied:
@kelvin-muchiri Yes we are directly using @opensrp/loader, but I've created this component as a wrapper which will work with the same props across all the UI and packages. Just to keep things in a place.

kelvin-muchiri commented on packages/keycloak-user-management/yarn.lock:
```diff
@@ -0,0 +1,16207 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Is this a new `yarn.lock` file? Should this be part of the changes?

kelvin-muchiri commented on packages/loader/README.md:
````diff
@@ -0,0 +1,47 @@
+## Installation
+
+```node
+yarn add @onaio/loaders
````
The README should be about `@opensrp/loader` and not `@onaio/loaders`. I think you should also mention the default props for the default props (configurations) used by OpenERP cc @p-netm @KipSigei

kelvin-muchiri commented on packages/loader/package.json:
```diff
@@ -0,0 +1,50 @@
+  "author": "OpenSRP Engineering",
+  "license": "Apache-2.0",
+  "private": false,
+  "dependencies": {
```
Seems like we do not require theses dependencies

kelvin-muchiri commented on packages/loader/package.json:
```diff
@@ -0,0 +1,50 @@
+    "@opensrp/server-service": "^0.0.11",
+    "@opensrp/store": "^0.0.6"
+  },
+  "devDependencies": {
```
Seems like we are not using this devDependencies too

kelvin-muchiri commented on packages/loader/package.json:
```diff
@@ -0,0 +1,50 @@
+    "@types/geojson": "^7946.0.7"
+  },
+  "peerDependencies": {
+    "@opensrp/loader": "^0.0.6",
```
Remove `@opensrp/loader` from peerDependencies

kelvin-muchiri commented on packages/loader/src/Loader.tsx:
```diff
@@ -0,0 +1,32 @@
+import { Ripple } from '@onaio/loaders';
+
+/** interface for Ripple props */
+export interface LoaderProps {
```
@p-netm @KipSigei @dubdabasoduba Should we really have this component configurable? I was thinking the `opensrp/loader` should not allow configuration so that the loader can be standard throughout the OpenERP package?

ashfahan replied:
i do agree with you, we shouldn't make this component configurable

p-netm replied:
@kelvin-muchiri @ashfahan what's stopping us from using it in standardized way if its configurable.

kelvin-muchiri replied:
@p-netm Say I import the package in user-management, I should not be allowed to pass my own configuration which will be different from location-management. i.e The loader should be exactly the same across the across components, across other packages

p-netm replied:
In embodiment of the principles that we wish the codebase to have. I do not think we can justify saying that any piece of code shouldn't be configurable, especially given that other devs are the intended consumers of the piece of code in question. In my view in such a case we should set the defaults to reflect what we want the standard to be, but still keep things flexible.
however this is just my personal preference.

ashfahan replied:
@p-netm  if we really need a configurable packages why dont we direactly the loader packages from jstools? .. i mean it does have the default one set to ripple right ? we really dont need have an extra wrapper around jstools's one ...

whats the points of having this extra loader packages ? i thought we are using this one to have our own *static* configured one *just for this project only*

kelvin-muchiri replied:
@p-netm I also thought at first that we should leave it configurable. But again like in my earlier example, it also means rationally, my code should pass code review if I pass my own configuration when I use the loader in a component. i.e It would be valid to pass my own config different from other components

haseemVD commented:
Closing PR, 
As discussed with @dubdabasoduba and team, 
We will use loaders from AntD,  New [issue](https://github.com/OpenSRP/web/issues/173) created.
